packages/graphback-codegen-schema/src/definitions/schemaDefinitions.ts

```diff
@@ -173,31 +174,40 @@ export const buildFilterInputType = (modelType: GraphQLObjectType) => {
 }
 
 export const buildCreateMutationInputType = (modelType: GraphQLObjectType) => {
-  const inputTypeName = getInputTypeName(modelType.name, GraphbackOperationType.CREATE);
+  return buildMutationInputFields(modelType, GraphbackOperationType.CREATE);
+}
+
+export const buildMutationInputType = (modelType: GraphQLObjectType) => {
+  return buildMutationInputFields(modelType, GraphbackOperationType.UPDATE);
+}
 
-  const idField = getPrimaryKey(modelType)
+function buildMutationInputFields(modelType: GraphQLObjectType, operation: GraphbackOperationType) {
+  const inputTypeName = getInputTypeName(modelType.name, operation);
+  const idField = getPrimaryKey(modelType);
   const allModelFields = getModelInputFields(modelType);
+  const fields = () => allModelFields
+  .reduce((fieldObj: any, field: GraphQLInputField) => {
+    const { name, type } = field
+    let fieldType: GraphQLNamedType;
+    let defaultValue: any;
```

Comment:
We could start creating interfaces from here on for new annotations and eventually add ones for old annotations at some point. What you think?

```suggestion
    let defaultValue: DefaultMetadata;
```

Or...Is it overkill to create a class instead which extends a BaseAnnotation.

```
const default = new DefaultAnnotation(field);
default.getValue();
```

Maybe a separate less pressing task, but this could make our annotations really maintainable in future. All could be in `graphback-core/src/annotations`

Reply:
> We could start creating interfaces from here on for new annotations and eventually add ones for old annotations at some point. What you think?

Good idea.

> Or...Is it overkill to create a class instead which extends a BaseAnnotation.
> 
> ```
> const default = new DefaultAnnotation(field);
> default.getValue();
> ```
> 
> Maybe a separate less pressing task, but this could make our annotations really maintainable in future. All could be in `graphback-core/src/annotations`

+1 on making this a separate task. I'll add the `DefaultMetadata` first.